src/main/diagnostics/limaDarwin.ts

```diff
@@ -0,0 +1,47 @@
+import semver from 'semver';
+
+import { spawnFile } from '@/utils/childProcess';
+import Logging from '@/utils/logging';
+
+import type { DiagnosticsCategory, DiagnosticsChecker } from './diagnostics';
+
+const console = Logging.diagnostics;
+
+/**
+ * CheckLimaDarwin version checks for compatibility with the host machine.
+ */
+const CheckLimaDarwin: DiagnosticsChecker = {
+  id:       'LIMA_DARWIN_VERSION',
+  category: 'General' as DiagnosticsCategory,
+  applicable() {
+    const isDarwin = process.platform === 'darwin';
+    const isArm = process.arch.startsWith('arm');
+
+    return Promise.resolve(isDarwin && isArm);
+  },
+  async check() {
+    const result = {
+      documentation: 'path#lima_darwin_version',
+      description:   '',
+      passed:        false,
+      fixes:         [] as { description: string }[],
+    };
+    const { stdout } = await spawnFile('sw_vers', ['-productVersion'], { stdio: ['ignore', 'pipe', console] });
+    let currentVersion = stdout.trim();
+
+    if (!/^\d+\.\d+\.\d+/.test(currentVersion)) {
+      currentVersion += '.0'; // massage it into looking like semver
+    }
+    result.passed = semver.gte(currentVersion, '12.4.0', { loose: true });
```

Comment:
This diagnostic should only fail if the VM is configured to use more than 3GB of memory. Otherwise there is no problem.

Reducing the VM size to 3GB would therefore also be a fix, but upgrading your OS is obviously preferable.

Reply:
Yep, which is why this is draft — #2908 will make it easier to look up the configuration.